Parser/myreadline.c

```diff
@@ -344,7 +344,7 @@ PyOS_StdioReadline(FILE *sys_stdin, FILE *sys_stdout, const char *prompt)
             break;
         }
         n += strlen(p + n);
-    } while (p[n-1] != '\n');
+    } while (n == 0 || p[n-1] != '\n');
```

Comment:
I would prefer to stop the loop if n==0.


```suggestion
    } while (n != 0 && p[n-1] != '\n');
```

Reply:
Then input like `b'\x00spam\nham\nspam\n'` will stop early. Is this intended?

Reply:
I'm not sure which behavior is the correct behavior honestly.

I suppose that you should call readline again to read `spam\n`.

Reply:
It sounds worth it to add a test at least.

Reply:
Then behavior will be different if string starts with NUL byte or has it in the middle. I think we should document this difference.

Reply:
We already have this conversation here: https://github.com/python/cpython/pull/140910#discussion_r2539490335.
Current fix seems to be slightly better for me.